Compile Error

Hello,
I am trying to build the dlt-daemon project for version 2.18.9, but I am getting the following compilation error.

Is there any way to resolve this?

My build environment looks like this

NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"

```
In file included from /usr/include/string.h:495,
                 from /home/dhjeong/dlt-daemon-2.18.9/src/console/dlt-control-common.c:56:
In function ‘strncpy’,
    inlined from ‘set_ecuid’ at /home/dhjeong/dlt-daemon-2.18.9/src/console/dlt-control-common.c:126:17:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: ‘__builtin_strncpy’ specified bound 10 equals destination size [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/console/logstorage/CMakeFiles/dlt-logstorage-ctrl.dir/build.make:102: src/console/logstorage/CMakeFiles/dlt-logstorage-ctrl.dir/__/dlt-control-common.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:599: src/console/logstorage/CMakeFiles/dlt-logstorage-ctrl.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/src/console/dlt-control-common.c
+++ b/src/console/dlt-control-common.c
@@ -124,9 +124,9 @@ void set_ecuid(char *ecuid)
             if (dlt_parse_config_param("ECUId", &ecuid_conf) == 0) {
                 memset(local_ecuid, 0, DLT_CTRL_ECUID_LEN);
                 strncpy(local_ecuid, ecuid_conf, DLT_CTRL_ECUID_LEN);
+                local_ecuid[DLT_CTRL_ECUID_LEN -1] = '\0';
                 if (ecuid_conf !=NULL)
                     free(ecuid_conf);
-                local_ecuid[DLT_CTRL_ECUID_LEN - 1] = '\0';
             }
             else {
                 pr_error("Cannot read ECUid from dlt.conf\n");

